pilot/pkg/security/authz/model/model.go

```diff
@@ -265,3 +272,42 @@ func (m *Model) Generate(service *ServiceMetadata, forTCPFilter bool) *envoy_rba
 	}
 	return policy
 }
+
+// getPrincipalsIncludingAliases returns a list of users encoded in SPIFFE format with the local
+// trust domain and its aliases.
+// For example, for a user "bar" in namespace "foo".
+// If the local trust domain is "td2" and its alias is "td1" (migrating from td1 to td2),
+// getPrincipalsIncludingAliases returns ["td2/ns/foo/sa/bar", "td2/ns/foo/sa/bar]].
+func getPrincipalsIncludingAliases(trustDomain string, trustDomainAliases []string, principals []string) []string {
+	// If trust domain aliases are empty, return the existing principals.
+	if len(trustDomainAliases) == 0 {
+		return principals
+	}
+	principalsIncludingAliases := []string{}
+	for _, principal := range principals {
+		trustDomainFromPrincipal := extractTrustDomainFromAuthzPrincipal(principal)
+		if trustDomainFromPrincipal == "" {
+			return principals
+		}
+		// Only generate configuration if the extracted trust domain from the policy is part of the trust domain aliases,
+		// or if the extracted/existing trust domain is "cluster.local", which is a pointer to the local trust domain
+		// and its aliases.
+		if !found(trustDomainFromPrincipal, trustDomainAliases) && trustDomainFromPrincipal != "cluster.local" {
+			continue
```

Comment:
Instead of using negative logic and continue, can you use positive logic and just include the actual logic here without using continue?